realm/realm-library/src/main/cpp/io_realm_RealmApp.cpp

```diff
@@ -27,6 +28,58 @@ using namespace realm::app;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
+// Helper method for constructing callbacks for REST calls that must return an actual result to Java
+template<typename T>
+std::function<void(T, Optional<app::AppError>)> create_result_callback(JNIEnv* env, jobject j_callback, const std::function<jobject (JNIEnv*, T)>& success_mapper) {
+    jobject callback = env->NewGlobalRef(j_callback);
+    return [callback, success_mapper](T result, Optional<app::AppError> error) {
```

Comment:
```suggestion
    return [&](T result, Optional<app::AppError> error) {
```
why not capture by reference?